platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/transaction/system/SystemTransactionExtractionUtils.java

```diff
@@ -71,12 +78,27 @@ private SystemTransactionExtractionUtils() {}
 
         final Iterator<Transaction> transactionIterator = event.transactionIterator();
         while (transactionIterator.hasNext()) {
+            final List<Bytes> transactions = event.getGossipEvent().transactions();
+            final boolean isNewFormat = !transactions.isEmpty();
             final Transaction transaction = transactionIterator.next();
-            if (systemTransactionTypeClass.isInstance(
-                    transaction.getTransaction().transaction().value())) {
-                scopedTransactions.add(
-                        new ScopedSystemTransaction<>(event.getCreatorId(), event.getSoftwareVersion(), (T)
-                                transaction.getTransaction().transaction().value()));
+            if (isNewFormat) {
+                if (systemTransactionTypeClass.equals(StateSignatureTransaction.class)) {
+                    try {
+                        final var maybeStateSignatureTransaction =
```

Comment:
The `Bytes` inside transaction will represent a whole `Transaction` proto type, where the StateSignatureTransaction body will be a nested field inside. This parsing logic should be reworked.

Reply:
Fixed